spec/unit/provider/oneview_logical_downlink_synergy_spec.rb

```diff
@@ -50,31 +59,38 @@
     end
 
     it 'should be able to find the resource' do
-      test = resourcetype.new(@client, resource['data'])
-      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
-      provider.exists?
       expect(provider.found).to be
     end
+  end
 
-    it 'should be able to get all the logical downlinks without ethernet' do
-      test = resourcetype.new(@client, resource['data'])
-      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
-      provider.exists?
-      allow_any_instance_of(resourcetype).to receive(:get_without_ethernet).and_return(test)
-      expect(provider.get_without_ethernet).to be
-      allow(resourcetype).to receive(:get_without_ethernet).with(anything).and_return('')
-      expect(provider.get_without_ethernet).to be
+  context '#get without ethernet', if: login[:api_version] >= 300 do
+    let(:resource) do
+      Puppet::Type.type(:oneview_logical_downlink).new(
+        name: 'Logical Downlink',
+        ensure: 'get_without_ethernet',
+        data:
+            {
+              'name' => 'Logical Downlink'
+            },
+        provider: 'synergy'
+      )
     end
 
-    it 'should be able to get logical downlinks without ethernet' do
-      test = resourcetype.new(@client, resource['data'])
-      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+    let(:test) { resourcetype.new(@client, resource['data']) }
+
+    before(:each) do
+      allow(resourcetype).to receive(:find_by).and_return([test])
       provider.exists?
-      allow_any_instance_of(resourcetype).to receive(:get_without_ethernet).and_return(test)
-      expect(provider.get_without_ethernet).to be
-      resource['data'] = {}
-      allow(resourcetype).to receive(:get_without_ethernet).with(anything).and_return(test)
-      expect(provider.get_without_ethernet).to be
     end
+
+    it 'should raise error when running get without ethernet' do
+      expect { provider.get_without_ethernet }.to raise_error(/The method #get_without_ethernet is unavailable for this resource/)
+    end
+
+    it 'should raise error when running get without ethernet - Test disabled due to bug on oneview-sdk'
+    # resource['data'] = {}
```

Comment:
'Test disabled due to bug on oneview-sdk' - Should be re-enabled/uncommented once oneview-sdk is fixed